arc_cache_test.go

```diff
@@ -45,6 +63,82 @@ func untrap(cd *callbackDatastore) {
 	cd.SetFunc(func() {})
 }
 
+type storeThrasher struct {
+	store *arccache
+
+	trace []blocks.Block
+
+	numBlocks int
+	numThreads int
+
+
+	ctx context.Context
+	cancel context.CancelFunc
+}
+
+func NewThrasher(store *arccache, numBlocks, numThreads int) (*storeThrasher, []blocks.Block) {
+	t :=  &storeThrasher{
+		numBlocks: numBlocks,
+		numThreads: numThreads,
+		store:      store,
+	}
+	trace := make([]blocks.Block, t.numBlocks)
+	for i := 0; i < t.numBlocks; i++ {
+		token := make([]byte, 4)
+		rand.Read(token)
+		trace[i] = blocks.NewBlock(token)
+	}
+	t.trace = trace
+	t.ctx, t.cancel = context.WithCancel(context.Background())
+
+	return t, trace
+}
+
+func (t *storeThrasher) Destroy() {
+	t.cancel()
+	t.store = nil
+}
+
+func (t *storeThrasher) Start() {
+	for i := 0; i < t.numThreads; i++ {
+		go func() {
+			for {
+				select {
+				case <-t.ctx.Done():
+					return
+				default:
+					idx := rand.Intn(t.numBlocks - 1)
```

Comment:
Suggestion: give each goroutine it's own `rand.Source`. The package level functions like `rand.Intn` use a shared source with a mutex.